backend/src/routes/api/images/imageUtils.ts

```diff
@@ -207,7 +216,14 @@ export const postImage = async (
   const customObjectsApi = fastify.kube.customObjectsApi;
   const namespace = fastify.kube.namespace;
   const body = request.body as BYONImageCreateRequest;
-  const imageTag = body.url.split(':')[1];
+  const fullUrl = body.url;
+  const matchArray = fullUrl.match(imageUrlRegex);
+  // check if the host is valid
+  if (!matchArray[1]) {
+    fastify.log.error('Invalid repository URL unable to add notebook image');
+    return { success: false, error: 'Invalid repository URL: ' + fullUrl };
```

Comment:
I guess that's the way we had the endpoint configure, but I'm not a fan of returning a 200 with a failing state, rather than using the proper error code.

Reply:
Yeah, there is a lot of uncomfortable stuff with this area of the code -- I'm viewing the UI and noticing a lot of inconsistencies to the other parts of our app... we will need to rework this.